src/lib.rs

```diff
@@ -1042,6 +1059,75 @@ impl<const N: usize, T> CircularBuffer<N, T> {
         }
     }
 
+    /// Appends an element to the back of the buffer.
+    ///
+    /// # Panics
+    ///
+    /// If the buffer is full, the method will panic.
+    ///
+    /// # Examples
+    ///
+    /// ```should_panic
+    /// use circular_buffer::CircularBuffer;
+    ///
+    /// let mut buf = CircularBuffer::<3, char>::new();
+    ///
+    /// buf.push_back_or_panic('a'); assert_eq!(buf, ['a']);
+    /// buf.push_back_or_panic('b'); assert_eq!(buf, ['a', 'b']);
+    /// buf.push_back_or_panic('c'); assert_eq!(buf, ['a', 'b', 'c']);
+    /// // The buffer is now full; adding more values causes panic
+    /// buf.push_back_or_panic('d');
+    /// ```
+    pub fn push_back_or_panic(&mut self, item: T) {
+        if N == 0 {
+            // Nothing to do
+            return;
+        }
+        if self.size >= N {
+            // At capacity; panic
+            self.panic_on_full("push_back");
+        } else {
+            // Some uninitialized slots left; append at the end
+            self.inc_size();
+            self.back_maybe_uninit_mut().write(item);
+        }
```

Comment:
I would rewrite this as just `self.try_push_back(item).expect("...")` to reduce the amount of code

Reply:
Done